etcdserver/server.go

```diff
@@ -777,7 +777,55 @@ func (s *EtcdServer) triggerSnapshot(ep *etcdProgress) {
 
 // Stop stops the server gracefully, and shuts down the running goroutine.
 // Stop should be called after a Start(s), otherwise it will block forever.
+// When stopping leader, Stop transfers its leadership to one of its peers
+// before stopping the server.
 func (s *EtcdServer) Stop() {
+	if s.Cfg != nil && s.Cfg.AutoLeaderTransfer &&
+		s.cluster != nil && len(s.cluster.MemberIDs()) > 1 &&
+		uint64(s.ID()) == s.Lead() {
+		transferee := uint64(0)
+		for _, id := range s.cluster.MemberIDs() {
+			if id != s.id {
+				transferee = uint64(id)
+				break
+			}
+		}
+		if transferee != 0 {
+			plog.Printf("stopping transport at %s", s.ID())
+			s.r.transport.Stop()
```

Comment:
how is it sending the transfer message if its network connection with raft is stopped?